test/e2e/wait_util.go

```diff
@@ -77,3 +77,45 @@ func waitForUserAccStatusConditions(t *testing.T, client client.Client, namespac
 		return false, nil
 	})
 }
+
+func waitForDeletedUserAccount(t *testing.T, client client.Client, name string, namespace string) error {
+	return wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		user := &toolchainv1alpha1.UserAccount{}
+		if err := client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, user); err != nil {
+			if errors.IsNotFound(err) {
+				t.Logf("deleted user account '%s'", name)
+				return true, err
+			}
+			return false, err
+		}
+		return false, nil
+	})
+}
+
+func waitForDeletedUser(t *testing.T, client client.Client, name string, namespace string) error {
+	return wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		user := &toolchainv1alpha1.UserAccount{}
+		if err := client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, user); err != nil {
+			if errors.IsNotFound(err) {
+				t.Logf("deleted user '%s'", name)
+				return true, err
```

Comment:
I guess, here rather returning `err` we should return nil to complete polling immediately ??

Reply:
i wanted to return error so the test was more clear on what was being checked. however, if you would like me to return nil and the test not verify that the error is "IsNotFound" i can do that as well...

Reply:
hmm.. i've updated it to remove the error. i guess the method name is clear enough to describe what is going on. removed the checks from the test as well!